ts/sagas/wallet.ts

```diff
@@ -542,6 +552,17 @@ export function* watchWalletSaga(
     if (action.type === FETCH_WALLETS_REQUEST && pagoPaToken.isSome()) {
       yield fork(fetchWallets, pagoPaClient, walletToken, pagoPaToken.value);
     }
+
+    if (action.type === ADD_CREDIT_CARD_REQUEST && pagoPaToken.isSome()) {
```

Comment:
it looks like there are a bunch of actions that are only valid when `pagoPaToken` is defined, how do we make sure that these actions are only triggered/handled when we have a valid pagopa token? one option is to fork the saga that listens to those actions whenever we have a valid token and pass it as a parameter

Reply:
it's not clear what happens when `pagoPaToken` is empty - and also I'm not sure it makes sense to store the pagopa session token in the redux state, what about something like the following instead:

```ts
while(true) {
 const maybePagopaSessionToken = getPagoPaSessionToken(...)
 if(maybePagopaSessionToken.isSome()) {
   yield call watchPagoPaActions(maybePagopaSessionToken.value);
 }  
}
```

then `watchPagoPaActions` is a saga that watches the wallet actions, whenever it receives an error of invalid session token it returns - in this way, the `watchPagoPaActions` saga is sure to have a valid session token and it doesn't have to keep checking it out from the redux store

what do you think?

Reply:
The pagoPa token management has been slightly changed now to be part of a single saga ([`fetchWithTokenRefresh`](https://github.com/teamdigitale/italia-app/blob/6f190f0c9320d2672650dbfbc6dd5cd93a862c76/ts/sagas/wallet.ts#L137)), which is then used by the rest of the sagas. 

What you suggest would still confine the `pagoPaToken` to a single saga (the watcher) and would avoid storing the token on the store. The downside is that, if an API call in a saga (e.g. add a card, that makes multiple API calls) fails b/c the session expired, the whole saga would have to fail and, at that point, the token can be refreshed: halting the saga because the session expired, though, is probably not the best approach, since the saga could be resumed right after the session is restored (as is being done right now). 

An alternative solution that would prevent the pagoPA token from being stored on redux is making it an attribute of a `PagoPaClient` class, which would then internally take care of refreshing it as needed when a call is made. As already mentioned, though, this introduces some of the process logic (the whole refreshing thing) in the client, which is intended to only provide interfaces to make API calls available. 

Is there any other way you would recommend doing this? And, is there something I'm missing about your proposed solution?